WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.WebClients.Wasabi;
+
+namespace WalletWasabi.Legal
+{
+	public class LegalDocuments
+	{
+		public const string EmbeddedFileName = "LegalDocuments.txt";
+		public const string LegalFolderName = "Legal";
+		public const string AssetsFoldername = "Assets";
+		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), LegalFolderName, AssetsFoldername, EmbeddedFileName);
+		public string FilePath { get; }
+		public Version Version { get; }
+
+		public static LegalDocuments TryLoadAgreed(string dataDir)
+		{
+			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
+			IoHelpers.EnsureDirectoryExists(legalFolderPath);
+			var filePaths = Directory.EnumerateFiles(legalFolderPath, "*.txt", SearchOption.TopDirectoryOnly);
+
+			// If more than one file found, then something strange happened, delete the dir and start from zero.
+			if (filePaths.Count() > 1)
+			{
+				IoHelpers.CleanDirectory(legalFolderPath);
+				filePaths = Enumerable.Empty<string>();
+			}
+
+			var existingFilePath = filePaths.FirstOrDefault();
+			if (existingFilePath is { })
+			{
+				var verString = Path.GetFileNameWithoutExtension(existingFilePath);
+				if (Version.TryParse(verString, out Version version))
+				{
+					string filePath = existingFilePath;
+					return new LegalDocuments(version, filePath);
+				}
+				else
+				{
+					File.Delete(existingFilePath);
+				}
+			}
+
+			return null;
```

Comment:
Here we should always provide some Legal docs. What if wasabi is used in offline mode or temporary cannot connect. In that case the installer included legal doc should be agreed at least.

Reply:
I disagree: https://github.com/zkSNACKs/WalletWasabi/pull/2805#issuecomment-575685274